controllers/tektontasks_controller.go

```diff
@@ -175,6 +185,94 @@ func (r *tektonTasksReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+func (r *tektonTasksReconciler) checkDeployedCRs(ctx context.Context, tektonRequest *common.Request) (bool, error) {
+	tektonTasksCRList, err := r.getCRList(ctx)
+	if err != nil {
+		tektonRequest.Logger.Error(err, "CRS listing failed")
+		return true, err
+	}
+
+	if len(tektonTasksCRList.Items) > 1 {
+		err := fmt.Errorf("there are multiple CRs deployed")
+		tektonRequest.Logger.Error(err, err.Error())
+
+		for _, tektonTask := range tektonTasksCRList.Items {
+			if tektonTask.Name != tektonRequest.Name {
+				degraded := false
+				for _, condition := range tektonTask.Status.Conditions {
+					if condition.Type == conditionsv1.ConditionDegraded {
+						degraded = condition.Status == v1.ConditionTrue
+						break
+					}
+				}
+
+				if !degraded {
+					namespacedName := types.NamespacedName{
+						Namespace: tektonTask.Namespace,
+						Name:      tektonTask.Name,
+					}
+					tektonTasks := &tekton.TektonTasks{}
+
+					errGet := r.client.Get(ctx, namespacedName, tektonTasks)
+					if errGet != nil {
+						return true, errGet
+					}
+
+					_, errSetStatus := r.setStatusMultipleCRs(tektonTasks, ctx, err)
+					if errSetStatus != nil {
+						return true, errSetStatus
+					}
+				}
+			}
+		}
+		return false, err
+	}
+	return false, nil
+}
+
+func (r *tektonTasksReconciler) getCRList(ctx context.Context) (*tekton.TektonTasksList, error) {
+	tektonTasksCRList := &tekton.TektonTasksList{}
+	tektonTasksCRList.SetGroupVersionKind(schema.GroupVersionKind{
```

Comment:
Is `SetGroupVersionKind` really required for the type `tekton.TektonTasksList`?

See this [example](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client@v0.11.1#example-Client-List).